src/t_zset.c

```diff
@@ -3555,19 +3561,23 @@ void zrangeGenericCommand(zrange_result_handler *handler, int argc_start, int st
     long opt_end = 0;
     int opt_withscores = 0;
     long opt_offset = 0;
-    long opt_limit = -1;
+    long opt_limit = -1; /* -1 means all. */
```

Comment:
No need to set since we have a flag (`limit_is_set`) telling us if this is initialized.
Also lets stay with the existing logic where any negative value means all.
```suggestion
    long opt_limit; /* negative means all. */
```